mmdet/core/general_data/general_data.py

```diff
@@ -0,0 +1,498 @@
+        return getattr(self, name)
+
+    def get(self, *args):
+        assert len(args) < 3, '`get` get more than 2 arguments'
+        return self.__dict__.get(*args)
+
+    def pop(self, *args):
+        assert len(args) < 3, '`pop` get more than 2 arguments'
+        name = args[0]
+        if name in self._meta_info_fields:
+            raise KeyError(f'{name} is a key in meta information, '
+                           f'which is immutable')
+
+        if args[0] in self._data_fields:
+            self._data_fields.remove(args[0])
+            return self.__dict__.pop(*args)
+
+        # with default value
+        elif len(args) == 2:
+            return args[1]
+        else:
+            raise KeyError(f'{args[0]}')
+
+    def __contains__(self, item):
+        return item in self._data_fields or \
+                    item in self._meta_info_fields
+
+    # Tensor-like methods
+    def to(self, *args, **kwargs):
+        """Apply same name function to all tensors in data_fields."""
+        new_data = self.new()
+        for k, v in self.items():
+            if hasattr(v, 'to'):
+                v = v.to(*args, **kwargs)
+            new_data[k] = v
+        return new_data
+
+    # Tensor-like methods
+    def cpu(self):
+        """Apply same name function to all tensors in data_fields."""
+        new_data = self.new()
+        for k, v in self.items():
+            if isinstance(v, torch.Tensor):
+                v = v.cpu()
+            new_data[k] = v
+        return new_data
+
+    # Tensor-like methods
+    def cuda(self):
+        """Apply same name function to all tensors in data_fields."""
+        new_data = self.new()
+        for k, v in self.items():
+            if isinstance(v, torch.Tensor):
+                v = v.cuda()
+            new_data[k] = v
+        return new_data
+
+    # Tensor-like methods
+    def detach(self):
+        """Apply same name function to all tensors in data_fields."""
+        new_data = self.new()
+        for k, v in self.items():
+            if isinstance(v, torch.Tensor):
+                v = v.detach()
+            new_data[k] = v
+        return new_data
+
+    # Tensor-like methods
+    def numpy(self):
+        """Apply same name function to all tensors in data_fields."""
+        new_data = self.new()
+        for k, v in self.items():
+            if isinstance(v, torch.Tensor):
+                v = v.detach().cpu().numpy()
+            new_data[k] = v
+        return new_data
+
+    def __nice__(self):
+        repr = '\n \n  META INFORMATION \n'
+        for k, v in self.meta_info_items():
+            repr += f'{k}: {v} \n'
+        repr += '\n   DATA FIELDS \n'
+        for k, v in self.items():
+            if isinstance(v, (torch.Tensor, np.ndarray)):
+                repr += f'shape of {k}: {v.shape} \n'
+            else:
+                repr += f'{k}: {v} \n'
+        return repr + '\n'
+
+
+class InstanceData(GeneralData):
+    """Subclass of :class:`GeneralData`.
```

Comment:
1. ```Subclass of :class:`GeneralData`.``` is not a good summary, may use `Data structure for instance-level annnotations or predictions` and put ```Subclass of :class:`GeneralData`.``` as description.